components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/resources/org.wso2.carbon.identity.organization.management.yaml

```diff
@@ -906,6 +1160,159 @@ components:
           type: string
           description: Shared application residing organization id.
           example: '682edf68-4835-4bb8-961f-0a16bc6cc866'
+
+    OrganizationDiscoveryPostRequest:
+      type: object
+      required:
+        - organizationId
+        - attributes
+      properties:
+        organizationId:
+          type: string
+          description: The ID of the organization whose discovery attributes are to be added.
+          example: '06c1f4e2-3339-44e4-a825-96585e3653b1'
+        attributes:
+          type: array
+          items:
+            $ref: '#/components/schemas/DiscoveryAttribute'
+    OrganizationDiscoveryAttributes:
+      type: object
+      required:
+        - attributes
+      properties:
+        attributes:
+          type: array
+          items:
+            $ref: '#/components/schemas/DiscoveryAttribute'
+    DiscoveryAttribute:
+      type: object
+      required:
+        - type
+        - value
+      properties:
+        type:
+          type: string
+          example: 'emailDomain'
+        values:
+          type: array
+          items:
+            type: string
+            example: 'abc.com'
+    OrganizationsDiscoveryResponse:
+      type: object
+      properties:
+        totalResults:
+          type: integer
+          example: 10
+        startIndex:
+          type: integer
+          example: 1
+        count:
+          type: integer
+          example: 10
+        links:
+          type: array
+          items:
+            $ref: '#/components/schemas/Link'
+          example:
+            [
+              {
+                "href": "/o/10084a8d-113f-4211-a0d5-efe36b082211/api/server/v1/organizations/discovery?filter=attributes.type+eq+emailDomain&limit=10&offset=50",
+                "rel": "next",
+              },  {
+              "href": "/o/10084a8d-113f-4211-a0d5-efe36b082211/api/server/v1/organizations/discovery?filter=attributes.type+eq+emailDomain&limit=10&offset=30",
+              "rel": "previous",
+            }
```

Comment:
do we provide such data in other index based paginated resources ?

Reply:
yes. ex: https://github.com/wso2/identity-api-server/blob/be5e9ca66df8ff260ab9bd5463d7e7eafcbbd767/components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/resources/idp.yaml#L2599-L2612